### ISSUE TITLE
Fix panic when disabling http server

### ISSUE THREAD

**Issue by labkode** [cs3org/reva#163]
```
6:21PM WRN grace/grace.go:172 > error reading pidfile pid=24835 pkg=grace
6:21PM INF grace/grace.go:181 > pidfile written to gateway.pid pid=24835 pkg=grace
6:21PM INF main.go:88 > running on 4 cpus pid=24835
6:21PM INF grpcserver/grpcserver.go:239 > chainning grpc unary interceptor log with priority 200 pid=24835 pkg=grpcserver
6:21PM INF grpcserver/grpcserver.go:268 > chainning grpc streaming interceptor log with priority 200 pid=24835 pkg=grpcserver
panic: runtime error: index out of range

goroutine 14 [running]:
main.main.func2(0xc000244a90, 0x1, 0x1, 0xc000244ac0, 0x1, 0x1, 0xc00008cc00, 0xc000300240)
        /home/gonzalhu/Development/reva/cmd/revad/main.go:126 +0x14d
created by main.main
        /home/gonzalhu/Development/reva/cmd/revad/main.go:125 +0x3dc

```

**Comment by unixlab** [comment]
I looked into this and I think I understood the problem.

At `reva/cmd/revad` from line 118 to 125 the code assume that the arrays `servers` and `listeners` have at least two entries ( index 0 as http server and index 1 as grpc server ):
```
if !coreConf.DisableGRPC {
	go func() {
		if err := servers[1].(*grpcserver.Server).Start(listeners[1]); err != nil {
			log.Error().Err(err).Msg("error starting the grpc server")
			watcher.Exit(1)
		}
	}()
}
```
But if the http server is disabled, we only have one entry ( index 0 ) and illegally access index 1.

I assumed that `servers` and `listeners` are arrays because it's possible that there will be a third, fours and maybe fives server one day.

I tried a few ways to fix this and submitted my final solution as pull request => https://github.com/cs3org/reva/pull/188
This change is bigger than I initially thought it would be, but this should be safe even with more services.
